dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java

```diff
@@ -40,6 +40,26 @@
      */
     private static final long serialVersionUID = 1L;
 
+    /*
+    * Returns the page mode based on the login mode or the FE/BE roles
+     */
+    @VisibleForTesting
+    public static PageMode processPageMode (final User user, final HttpServletRequest request) {
+
+        final LoginMode loginMode = LoginMode.get(request);
+
+        if (LoginMode.UNKNOWN == loginMode) {
+
+            return user.isFrontendUser()
+                    ? PageMode.setPageMode(request, PageMode.LIVE)
+                    :  user.isBackendUser()
+                    ? PageMode.getWithNavigateMode(request)
+                    :  PageMode.setPageMode(request, PageMode.LIVE);
+        }
+
+        return  LoginMode.FE == loginMode?
+                PageMode.setPageMode(request, PageMode.LIVE): PageMode.getWithNavigateMode(request);
+    }
```

Comment:
should this method be here?, maybe it should be move to PageMode

Reply:
I thought the same, but I am concern about breaking something